odbc/src/api/cancel_tests.rs

```diff
@@ -0,0 +1,140 @@
+    //             let mut query: Vec<WideChar> = cstr::to_widechar_vec("select * from batters where batters.Player in \
+    //             ( select b2.Player from batters b2 where b2.BB / b2.AB > ( SELECT avg(b3.BB / b3.AB) from batters b3 ))");
+    //             query.push(0);
+    //             assert_eq!(
+    //                 SQLExecDirectW(stmt_ref as *mut _, query.as_ptr(), query.len() as i32),
+    //                 SqlReturn::ERROR
+    //             );
+
+    //             // when SQLCancel is called below, the long running query should be cancelled; verify we return the proper error in this case
+    //             assert_eq!(
+    //                 "[MongoDB][Core] Query was cancelled".to_string(),
+    //                 format!(
+    //                     "{}",
+    //                     (*stmt_ref).as_statement().unwrap().errors.read().unwrap()[0]
+    //                 ),
+    //             );
+    //         });
+
+    //         // use SQLCancel to cancel the query
+    //         thread::sleep(time::Duration::from_secs(5)); // ensure query has time to be issued before cancelling
+    //         assert_eq!(SQLCancel(stmt as *mut _), SqlReturn::SUCCESS);
+
+    //         // verify the query is no longer running by checking no current operations have the statement id as a comment
+    //         thread::sleep(time::Duration::from_secs(5)); // ensure query has time to be cancelled before validating
+    //         let mut cursor = conn
+    //             .as_connection()
+    //             .unwrap()
+    //             .mongo_connection
+    //             .read()
+    //             .unwrap()
+    //             .as_ref()
+    //             .unwrap()
+    //             .client
+    //             .database("admin")
+    //             .aggregate(vec![doc! {"$currentOp": {}}], None)
+    //             .unwrap();
+
+    //         assert!(!cursor.any(|row| {
+    //             let operation = row.unwrap_or_default();
+    //             if let Some(bson::Bson::Document(d)) = operation.get("command") {
+    //                 d.get("comment") == Some(&stmt_id)
+    //             } else {
+    //                 false
+    //             }
+    //         }));
+    //     }
+    // }
+
+    #[test]
+    fn test_cancel_no_running_query() {
+        let env = &mut MongoHandle::Env(Env::with_state(EnvState::Allocated));
+        let conn =
+            &mut MongoHandle::Connection(Connection::with_state(env, ConnectionState::Allocated));
+
+        // StatementState::Allocated is the only valid state other than the state indicating a query is executing.
+        // SQLCancel should be a no op in this case; verify we get a success.
+        let stmt: *mut _ =
+            &mut MongoHandle::Statement(Statement::with_state(conn, StatementState::Allocated));
+        unsafe { assert_eq!(SQLCancel(stmt as *mut _), SqlReturn::SUCCESS) }
+    }
+
+    #[test]
+    fn test_cancel_running_query_not_executing() {
```

Comment:
Maybe add a comment here to say that this test is checking that if at the time Cancel is called the query is still executing, but when killOp is called, the execution is over, then we still gracefully return SUCCESS.